### PR TITLE
Deployment fix for `--auto_create_cli_app` flag bug

### LINKED ISSUE #2915: Deployment section of Getting Started guide is out of date
The deployment steps don't give any guidance on how to generate the client_id code for the config file.

I was able to get a deployment set up in release 6.3.0 by adding the --auto_create_cli_app flag to the deploy.py arguments, but with the refactoring for the config file for 6.4.0 the same command fails with:
"Exception: CLI client_id not provided as valid key. Please Provide Valid Config."

[AB#36144](https://fuzzfest.visualstudio.com/5eaacf23-2084-4d62-8fbe-7d37326a8b74/_workitems/edit/36144)

### PR DESCRIPTION
**Pull request by AdamL-Microsoft**
## Summary of the Pull Request

Fixes #2915.

Refactoring the deployment config and OneFuzz cli config settings created a bug that required new deployments know the CLI app ID prior to it being created in situations where the CLI app ID is specifically set to be created at deploy time using the `--auto_create_cli_app` flag.

## PR Checklist
* [x] Applies to work item: #2915
* [x] Tests added/passed
* [x] Requires documentation to be updated

## Info on Pull Request

updated:
src/deployment/deploy.py
src/deployment/deploylib/configuration.py

## Validation Steps Performed
* run `deploy.py $REGION $RESOURCE_GROUP $APP_NAME $OWNER config.json --auto_create_cli_app`  (where config.json doesn't have the normally required "cli_client_id" key)

### CODE REVIEW

**Review by nharper285 — COMMENTED** [review]
This looks good. 

Have you tested in PME _and_ microsoft tenant? 

The test cases I would think about are: 
- [ ] Deploying to Microsoft using the Micrsoft onefuzz-cli app id in `config.json`
- [ ] Using the `auto_create` flag to create a cli app id in Microsot tenant
- [ ] Deploying to PME using the PME onefuzz-cli app id in `config.json`
- [ ] Using the `auto_create flag to create a cli app id in PME tenant.

File: docs/getting-started.md [hunk 1]
```diff
@@ -40,7 +40,7 @@ chmod +x deploy.py
 When running `deploy.py` the first time for an instance, you will be prompted
 to follow a manual step to initialize your CLI config.
 
-The `$NSG_CONFIG_FILE` is a required parameter that specifies the 'allow rules' for the OneFuzz Network Security Group. A default `config.json` is provided in the deployment zip.
+The `$NSG_CONFIG_FILE` is a required parameter that specifies the 'allow rules' for the OneFuzz Network Security Group as well as other basic OneFuzz settings. A default `config.json` is provided in the deployment zip.
```
[the remark this answers is not in the document]
Comment (AdamL-Microsoft) [reply docs/getting-started.md:43]: Great advice - This has been tested in the MS tenant in those situations several times each, I haven't tested it on the PME tenant yet at all. I'll run through the test cases you've listed out and see.

**Comment by AdamL-Microsoft** [comment]
> This looks good.
> 
> Have you tested in PME _and_ microsoft tenant?
> 
> The test cases I would think about are:
> 
> * [x]  Deploying to Microsoft using the Micrsoft onefuzz-cli app id in `config.json`
> * [x]  Using the `auto_create` flag to create a cli app id in Microsot tenant
> * [x]  Deploying to PME using the PME onefuzz-cli app id in `config.json`
> * [x]  Using the `auto_create flag to create a cli app id in PME tenant.

These tests completed successfully, details in teams.

**Review by mgreisen — APPROVED** [review]
:shipit:

**Review by nharper285 — APPROVED** [review]

**Pull request merged**